packages/eslint-plugin/lib/configs/style.js

```diff
@@ -47,6 +50,10 @@ module.exports = {
     'jsdoc/require-yields': 'off',
     'jsdoc/tag-lines': 'off',
     'jsdoc/valid-types': 'off',
+    'jsdoc/check-types': [
+      'warn',
+      { exemptTagContexts: [{ tag: 'typedef', types: ['Object'] }] },
```

Comment:
Your issue makes a good case for allowing `Object`. However, TypeScript [discourages `Object`](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#:~:text=%E2%9D%8C%20Don't%20ever%20use,used%20appropriately%20in%20JavaScript%20code.) 

Moreover, this is a "style" config so it's meant to be have opinions. Even insofar as they're interchangeable, using one consistently is worthwhile.

Reply:
It's true that TypeScript discourages `Object` _in TypeScript_, but the rules for JS usage are different, because TS maintains some modicum of compatibility with JSDoc.

If I had my way, all object-based `@typedefs` would just _omit_ `{Object}` or `{object}`, since it's implied.  In fact, we can do that without changing any configuration; it's not enforced consistently as-is.

This is likely a situation in which I am the only one really bothered by the way it is currently.  If the change seems unreasonable to others, I will close this.